ui/app/common/concept-set/directives/conceptSet.js

```diff
@@ -238,7 +238,8 @@ angular.module('bahmni.common.conceptSet')
                 };
 
                 var runFormConditionForAllObsRecursively = function (formName, rootObservation) {
-                    _.each(rootObservation.groupMembers, function (observation) {
+                    rootObservation && rootObservation.groupMembers &&
```

Comment:
Hi @maharjunm, Please change `rootObservation && rootObservation.groupMembers && rootObservation.groupMembers.forEach(function(){
....
)` to `rootObservation && _.each(rootObservation.groupMembers, function(){
....
})`

Reply:
Hi @sumanmaity112 
This itself is the fix. We explained in previous comments why we are migrating from _.each to .forEach.

Reply:
Hi @sai789, 
If I understood your comments, you need to check * rootObservation*  is `undefined` or not.
You can do that using the following way as well
`rootObservation && _.each(rootObservation.groupMembers, function(){ .... })`. This same as `rootObservation && rootObservation.groupMembers && rootObservation.groupMembers.forEach(function(){ .... )`

Reply:
Hi @sumanmaity112, 
We have replaced _.each with forEach because  _.each does not adhere to concurrent modification of iterating list. Using forEach that works for concurrent modification.
This change is the fix